Need version of metro-wsit / metro.webservices-api-osgi supporting jakarta.servlet 6

Integrating metro 4.0.0-M1 in GF 7 leads to several exceptions such as

```
 java.lang.NoClassDefFoundError: jakarta/servlet/ServletContainerInitializer.
```

Upon investigation of the MANIFEST.MF, it seems a number of Jakarta dependencies are restricted to their EE 9.1 versions:

```
      jakarta.security.auth.message.callback;resolution:=optional;version="[2.0,3)"
      jakarta.security.auth.message.config;resolution:=optional;version="[2.0,3)"
      jakarta.security.auth.message.module;resolution:=optional;version="[2.0,3)"
      jakarta.security.auth.message;resolution:=optional;version="[2.0,3)"
      jakarta.servlet.annotation;resolution:=optional;version="[5.0,6)"
      jakarta.servlet.http;resolution:=optional;version="[5.0,6)"
      jakarta.servlet;resolution:=optional;version="[5.0,6)"
```

GlassFish 7 bundles Jakarta Authentication 3 and Jakarta Servlet 6 among others.

I think a new version of "org.glassfish.metro.webservices-api-osgi" is needed?

wrt issue in grizzly:

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project webservices-extra: Compilation failure: Compilation failure: 
[ERROR] /Users/lukas/development/github/eclipse-ee4j/metro-wsit/wsit/bundles/webservices-extra/target/generated-sources/dependencies/org/glassfish/grizzly/servlet/HttpServletRequestImpl.java:[83,8] org.glassfish.grizzly.servlet.HttpServletRequestImpl is not abstract and does not override abstract method getServletConnection() in jakarta.servlet.ServletRequest
[ERROR] /Users/lukas/development/github/eclipse-ee4j/metro-wsit/wsit/bundles/webservices-extra/target/generated-sources/dependencies/org/glassfish/grizzly/servlet/HttpServletRequestImpl.java:[696,5] method does not override or implement a method from a supertype
[ERROR] /Users/lukas/development/github/eclipse-ee4j/metro-wsit/wsit/bundles/webservices-extra/target/generated-sources/dependencies/org/glassfish/grizzly/servlet/HttpServletRequestImpl.java:[1040,5] method does not override or implement a method from a supertype
[ERROR] /Users/lukas/development/github/eclipse-ee4j/metro-wsit/wsit/bundles/webservices-extra/target/generated-sources/dependencies/org/glassfish/grizzly/servlet/HttpServletResponseImpl.java:[447,5] method does not override or implement a method from a supertype
[ERROR] /Users/lukas/development/github/eclipse-ee4j/metro-wsit/wsit/bundles/webservices-extra/target/generated-sources/dependencies/org/glassfish/grizzly/servlet/HttpServletResponseImpl.java:[455,5] method does not override or implement a method from a supertype
[ERROR] /Users/lukas/development/github/eclipse-ee4j/metro-wsit/wsit/bundles/webservices-extra/target/generated-sources/dependencies/org/glassfish/grizzly/servlet/HttpServletResponseImpl.java:[633,5] method does not override or implement a method from a supertype
[ERROR] /Users/lukas/development/github/eclipse-ee4j/metro-wsit/wsit/bundles/webservices-extra/target/generated-sources/dependencies/org/glassfish/grizzly/servlet/HttpSessionImpl.java:[161,5] method does not override or implement a method from a supertype
[ERROR] /Users/lukas/development/github/eclipse-ee4j/metro-wsit/wsit/bundles/webservices-extra/target/generated-sources/dependencies/org/glassfish/grizzly/servlet/HttpSessionImpl.java:[177,5] method does not override or implement a method from a supertype
[ERROR] /Users/lukas/development/github/eclipse-ee4j/metro-wsit/wsit/bundles/webservices-extra/target/generated-sources/dependencies/org/glassfish/grizzly/servlet/HttpSessionImpl.java:[193,5] method does not override or implement a method from a supertype
[ERROR] /Users/lukas/development/github/eclipse-ee4j/metro-wsit/wsit/bundles/webservices-extra/target/generated-sources/dependencies/org/glassfish/grizzly/servlet/HttpSessionImpl.java:[274,5] method does not override or implement a method from a supertype
[ERROR] /Users/lukas/development/github/eclipse-ee4j/metro-wsit/wsit/bundles/webservices-extra/target/generated-sources/dependencies/org/glassfish/grizzly/servlet/HttpSessionImpl.java:[323,5] method does not override or implement a method from a supertype
[ERROR] /Users/lukas/development/github/eclipse-ee4j/metro-wsit/wsit/bundles/webservices-extra/target/generated-sources/dependencies/org/glassfish/grizzly/servlet/WebappContext.java:[1250,5] method does not override or implement a method from a supertype
[ERROR] /Users/lukas/development/github/eclipse-ee4j/metro-wsit/wsit/bundles/webservices-extra/target/generated-sources/dependencies/org/glassfish/grizzly/servlet/WebappContext.java:[1261,5] method does not override or implement a method from a supertype
[ERROR] /Users/lukas/development/github/eclipse-ee4j/metro-wsit/wsit/bundles/webservices-extra/target/generated-sources/dependencies/org/glassfish/grizzly/servlet/WebappContext.java:[1273,5] method does not override or implement a method from a supertype
[ERROR] /Users/lukas/development/github/eclipse-ee4j/metro-wsit/wsit/bundles/webservices-extra/target/generated-sources/dependencies/org/glassfish/grizzly/servlet/WebappContext.java:[1292,5] method does not override or implement a method from a supertype
[ERROR] /Users/lukas/development/github/eclipse-ee4j/metro-wsit/wsit/bundles/webservices-extra/target/generated-sources/dependencies/org/glassfish/grizzly/servlet/SessionCookieConfig.java:[22,8] org.glassfish.grizzly.servlet.SessionCookieConfig is not abstract and does not override abstract method getAttributes() in jakarta.servlet.SessionCookieConfig
[ERROR] /Users/lukas/development/github/eclipse-ee4j/metro-wsit/wsit/bundles/webservices-extra/target/generated-sources/dependencies/org/glassfish/grizzly/servlet/DispatchedHttpServletResponse.java:[183,5] method does not override or implement a method from a supertype
[ERROR] /Users/lukas/development/github/eclipse-ee4j/metro-wsit/wsit/bundles/webservices-extra/target/generated-sources/dependencies/org/glassfish/grizzly/servlet/DispatchedHttpServletResponse.java:[188,14] method setStatus in class jakarta.servlet.http.HttpServletResponseWrapper cannot be applied to given types;
[ERROR]   required: int
[ERROR]   found: int,java.lang.String
[ERROR]   reason: actual and formal argument lists differ in length
[ERROR] -> [Help 1]
```

--- a/.github/workflows/maven.yml
+++ b/.github/workflows/maven.yml
@@ -70,6 +70,7 @@ jobs:
         with:
           distribution: 'zulu'
           java-version: ${{ needs.build.outputs.jdk }}
+          cache: maven
       - name: Checkout tests
         uses: actions/checkout@v2.3.4
         with:

--- a/wsit/boms/bom/pom.xml
+++ b/wsit/boms/bom/pom.xml
@@ -53,7 +53,7 @@
     </developers>
 
     <properties>
-        <jaxws.ri.version>4.0.0-M1</jaxws.ri.version>
+        <jaxws.ri.version>4.0.0-M2</jaxws.ri.version>
     </properties>
 
     <dependencyManagement>

